pkg/webhook/admission/inferenceservice/inferenceservice_validator.go

```diff
@@ -43,9 +48,36 @@ func (validator *Validator) Handle(ctx context.Context, req admissiontypes.Reque
 		return admission.ErrorResponse(http.StatusInternalServerError, err)
 	}
 
+	if constants.IsEnableWebhookNamespaceSelector {
+		if err := validator.validateNamespace(isvc, req.AdmissionRequest.Namespace); err != nil {
+			return admission.ErrorResponse(http.StatusBadRequest, err)
+		}
+	}
+
 	if err := isvc.ValidateCreate(validator.Client); err != nil {
 		return admission.ErrorResponse(http.StatusBadRequest, err)
 	}
 
 	return admission.ValidationResponse(true, "allowed")
 }
+
+func (validator *Validator) validateNamespace(isvc *kfserving.InferenceService, namespace string) error {
```

Comment:
Why do we need any of this validation code? As I understand it, this is executed by the k8s masters and configured via the webhook.

Reply:
nvm I get it. We want to prevent isvcs from being created if the pods can't be created. +1